File: management-domain/Digital-marketing/Backend/Java/seo-service/src/main/java/com/gogidix/digitalmarketing/shared/exception/NotFoundException.java
package com.gogidix.digitalmarketing.shared.exception;

/**
 * Exception thrown when a requested resource is not found.
 */
public class NotFoundException extends DomainException {

    public NotFoundException(String resource, String id) {
        super("NOT_FOUND", String.format("%s not found: %s", resource, id));
    }

    public NotFoundException(String message) {
        super("NOT_FOUND", message);
    }

    public NotFoundException(String resource, String id, Throwable cause) {
        super("NOT_FOUND", String.format("%s not found: %s", resource, id), cause);
    }
}

File: management-domain/Sales-department/Backend/Nodes/sales-automation-service/commission-calculator-service/src/shared/exceptions/not-found.exception.ts
import { HttpStatus } from '@nestjs/common';
import { BaseException } from './base.exception';

/**
 * Not found exception
 * Thrown when a requested resource is not found
 */
export class NotFoundException extends BaseException {
  readonly errorCode = 'NOT_FOUND';

  constructor(message: string, context?: Record<string, unknown>) {
    super(message, HttpStatus.NOT_FOUND, context);
  }
}
